core/src/network/actions/get_links.rs

```diff
@@ -17,11 +17,13 @@ use std::{pin::Pin, sync::Arc, thread};
 pub async fn get_links(
     context: Arc<Context>,
     address: Address,
-    tag: String,
+    link_type: String,
```

Comment:
Hm, I guess we also want this to be an Option
```suggestion
    link_type: Option<String>,
```
so that we could requests all links with the same tag but different types. That would match the `get_links` semantic we had in proto, no?
@zippy

Reply:
So you think we should make `link_type` optional as well? I guess this is possible but not sure if it makes sense...

I am also not sure if we should be forcing every link to be tagged. This is how it works currently but I suppose we could let people pass a `None` to `link_entries` and have the HDK add a sensible default behind the scenes. There are a few usability things to sort out.

Reply:
I guess it makes sense to have no tag, i.e. `None`.

It does make sense to have the type optional in the `get_links` signature so that we can requests all links with the same tag but from different types.

Imagine a use-case where agents can add all different kinds of comments on entries, each being a link with a tag `comment`. But the reason why that comment is a valid statement could differ which would best be modeled by having multiple link types with different validation rules. But for rendering all those comments a UI might not care about why something is a valid comment, not caring for the type but instead requesting any link with the tag `comment` independent of the type.